File: 01_two_sum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut res = Vec::<i32>::new();
        for i in 0..(nums.len() - 1) {
            for j in (i + 1)..nums.len() {
                if nums[i] + nums[j] == target {
                    res.push(i as i32);
                    res.push(j as i32);
                }
            }
        }
        res
    }
}

File: python/01_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # temp_list = []
        # for i in range(len(nums) - 1):
        #     for j in range(i + 1, len(nums)):
        #         if nums[i] + nums[j] == target:
        #             temp_list.append(i)
        #             temp_list.append(j)
        #             return temp_list

        hash_table = {}
        for i, num in enumerate(nums):
            remaining = target - num
            if remaining not in hash_table:
                hash_table[num] = i
            else:
                return [hash_table[remaining], i]

